src/main/java/fact/cleaning/TwoLevelTimeNeighbor.java

```diff
@@ -175,25 +171,26 @@ public Data process(Data input) {
 	 * @param minNumberOfNeighborPixel
 	 * @return
 	 */
-	public ArrayList<Integer> applyTimeNeighborCleaning(ArrayList<Integer> showerPixel,double[] arrivalTime, double timeThreshold, int minNumberOfNeighborPixel) {
+	public PixelSet applyTimeNeighborCleaning(PixelSet showerPixel, double[] arrivalTime, double timeThreshold, int minNumberOfNeighborPixel) {
+
+        PixelSet newShowerPixels = new PixelSet();
 
+        for(CameraPixel pixel: showerPixel){
 
-		ArrayList<Integer> newList= new ArrayList<Integer>();
-		for(int pixel: showerPixel){
-			FactCameraPixel[] currentNeighbors = pixelMap.getNeighboursFromID(pixel);
+        	FactCameraPixel[] currentNeighbors = pixelMap.getNeighboursForPixel(pixel);
```

Comment:
Inconsistent english: `applyTimeNeighborCleaning` but `pixelMap.getNeighboursForPixel` (I know it was not you but maybe you can change that on the go)